packages/react-instantsearch-core/src/core/createStore.ts

```diff
@@ -0,0 +1,38 @@
+type SearchState = any; // @TODO: give searchState type
+type ResultsState = any; // @TODO: give results type
+type ResultsFacetsValues = any;
+type Listener = () => any;
+type State = {
+  widgets: SearchState;
+  metadata: any[];
+  results: ResultsState | null;
+  resultsFacetValues: ResultsFacetsValues | null;
+  error: any | null;
+  searching: boolean;
+  isSearchStalled: boolean;
+  searchingForFacetValues: boolean;
+};
+export default function createStore(initialState: State) {
+  let state = initialState;
+  const listeners: Listener[] = [];
+  function dispatch() {
+    listeners.forEach(listener => listener());
+  }
+  return {
+    getState() {
```

Comment:
We should enforce the return type.

Reply:
I'm not sure what you mean, this code didn't change by the way